### PR TITLE
BHBC-1665: Populate project draft name with project name

### PR DESCRIPTION
**Pull request by curtisupshall**
# Overview

## Links to jira tickets

https://quartech.atlassian.net/browse/BHBC-1665

## This PR contains the following changes

Create/edit project: Save Draft  
 - Default the draft name field to the project title (if the user had entered one).
 - The user should still be able to overwrite it if they would rather name it something else.

### CODE REVIEW

File: app/src/features/projects/create/CreateProjectPage.tsx [hunk 1]
```diff
@@ -337,7 +338,9 @@ const CreateProjectPage: React.FC = () => {
         component={{
           element: <ProjectDraftForm />,
           initialValues: {
-            draft_name: '' // TODO
+            draft_name: formikRef.current
```
Comment (KjartanE) [inline app/src/features/projects/create/CreateProjectPage.tsx:341]: Would formikRef.current.values.project.project_name be more appropriate as the initial if statement?

Comment (curtisupshall) [reply app/src/features/projects/create/CreateProjectPage.tsx:341]: `formikRef.current` is possibly null, but if it's not null, then `formikRef.current.project` should be defined. But if we write `formikRef.current.values.project.project_name ?` then ESlint will still complain that "`forkmikRef.current` is possibly undefined."

Comment (NickPhura) [reply app/src/features/projects/create/CreateProjectPage.tsx:341]:
```
draft_name: formikRef.current?.values.project.project_name || ProjectDraftFormInitialValues.draft_name
```

**Review by KjartanE — APPROVED** [review]
Looks good!

**Review by NickPhura — APPROVED** [review]

**Pull request merged**